### PR TITLE
SE-779-Modify-Token-Level-Period-Based-Rule-Data-Accumulation

### PR DESCRIPTION
**Pull request by ShaneDuncan602**
…g Volume

### CODE REVIEW

**Review by TJ-Everett — APPROVED** [review]
This looks great Shane! It took me a minute to understand the changes here with the active flag and where that is stored now. Very clever!! Approved

File: src/client/token/handler/ruleContracts/HandlerTokenMaxSupplyVolatility.sol [hunk 1]
```diff
@@ -35,30 +36,13 @@ contract HandlerTokenMaxSupplyVolatility is RuleAdministratorOnly, ActionTypesAr
...
     /**
      * @dev Clear the rule data structure
      */
     function clearTokenMaxSupplyVolatility() internal {
         TokenMaxSupplyVolatilityS storage data = lib.tokenMaxSupplyVolatilityStorage();
-        for (uint i; i <= lib.handlerBaseStorage().lastPossibleAction;) {
+        delete data.ruleId;
+        for (uint i; i <= lib.handlerBaseStorage().lastPossibleAction; ++i) {
             delete data.tokenMaxSupplyVolatility[ActionTypes(i)];
             unchecked {
```
Comment (Palmerg4) [inline src/client/token/handler/ruleContracts/HandlerTokenMaxSupplyVolatility.sol]: The unchecked increment is still present

File: src/client/token/handler/ruleContracts/HandlerTokenMaxTradingVolume.sol [hunk 2]
```diff
@@ -36,30 +37,13 @@ contract HandlerTokenMaxTradingVolume is RuleAdministratorOnly, ActionTypesArray
...
     /**
      * @dev Clear the rule data structure
      */
     function clearTokenMaxTradingVolume() internal {
         TokenMaxTradingVolumeS storage data = lib.tokenMaxTradingVolumeStorage();
-        for (uint i; i <= lib.handlerBaseStorage().lastPossibleAction;) {
+        delete data.ruleId;
+        for (uint i; i <= lib.handlerBaseStorage().lastPossibleAction; ++i) {
             delete data.tokenMaxTradingVolume[ActionTypes(i)];
             unchecked {
```
Comment (Palmerg4) [inline src/client/token/handler/ruleContracts/HandlerTokenMaxTradingVolume.sol]: Here as well. Extra unchecked incrementation

**Review by Palmerg4 — CHANGES_REQUESTED** [review]
Just a couple unchecked increments I think aren't intended

File: src/client/token/handler/ruleContracts/HandlerTokenMaxSupplyVolatility.sol (hunk 1 above)
Comment (ShaneDuncan602) [reply src/client/token/handler/ruleContracts/HandlerTokenMaxSupplyVolatility.sol]: Good catch. This must've been a merge issue.

**Review by Palmerg4 — APPROVED** [review]
Look awesome, nice work!

**Pull request merged**